### PR TITLE
ci: run JVM&JS on Linux

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

PRs are taking to long to be merged.

### Causes

It takes ages to build native libs and run tests on all targets

### Solutions

- Use docker image with native libs to run JVM tests.
- Also run JS tests on docker/Linux, in order to spread a bit the tests across runners.
- Keep running Android and iOS tests on Mac runners.
- Add gradle and AVD caches

Sample runs: 
- [JVM & JS](https://github.com/wireapp/kalium/runs/5660716641)
- [Android & iOS](https://github.com/wireapp/kalium/runs/5660713960)

### Testing

CI is being tested on this PR.

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by vitorhugods** [comment]
Seems to be working and skiping the `build native libs` step: https://github.com/wireapp/kalium/runs/5648288857?check_suite_focus=true

**Review by yamilmedina — APPROVED** [review]
Nice work! Just a small comment (kind of lint related). BTW, great solution, splitting up the tests targets onto different containers 🪄

**Review by gongracr — APPROVED** [review]
Looking great! Can't wait to see how much are we reducing our CI building times 🤓

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,9 @@
 JAVA_HOME := $(shell /usr/libexec/java_home)
 CRYPTOBOX_C_VERSION := "v1.1.3"
 LIBSODIUM_VERSION := "1.0.18-RELEASE"
+ARTIFACTS_PATH := native/libs
```
[the remark this answers is not in the document]
Comment (vitorhugods) [reply Makefile:4]: Indeed. Remains from my original approach 🤪

**Pull request merged**